docs/libcudacxx/index.rst

```diff
@@ -96,11 +96,13 @@ Summary: ``std::``, ``cuda::`` and ``cuda::std::``
    can use the ``--expt-relaxed-constexpr`` flag to use any ``constexpr`` functions in ``__device__`` code.
    libcu++ does not replace or interfere with host compiler's Standard Library.
 -  ``cuda::std::`` / ``<cuda/std/*>``: Conforming implementations of facilities from the Standard Library that work in
-   ``__host__ __device__`` code.
--  ``cuda::`` / ``<cuda/*>``: Conforming extensions to the Standard Library that work in ``__host__ __device__`` code.
+   ``__host__`` and  ``__device__`` code.
+-  ``cuda::`` / ``<cuda/*>``: Conforming extensions to the Standard Library that work in ``__host__`` and ``__device__`` code.
 -  ``cuda::device`` / ``<cuda/device/*>``: Conforming extensions to the Standard Library that work only in
    ``__device__`` code.
--  ``cuda::ptx``: C++ convenience wrappers for inline PTX (only usable in ``__device__`` code).
+-  ``cuda::ptx``: C++ convenience wrappers for inline PTX that work only in ``__device__`` code.
+
+Any libcu++ feature that works in ``__host__`` code also works when being compiled by a non-CUDA compiler (e.g., GCC).
```

Comment:
I'm hesitant about documenting this until we add host-only compiler testing to our CI. See https://github.com/NVIDIA/cccl/issues/1658

Reply:
In a discussion with @wmaxey he pointed out that this is the guarantee we already try to give and if we fail it's a bug. So I would prefer documenting that clearly, since this information is otherwise hard to discover (I had to ask around) and severely limiting (libc++ couldn't be used in any codebase portable to compilers outside nvcc).

We do have partial test coverage for the libcu++ parts that Thrust pulls into the OMP and TBB backends. I agree that dedicated tests would be better.

If it makes you feel better, I can mark this PR as blocked on #1658 and wait.